OSX build error does not fail Travis

From https://github.com/mapbox/mapbox-gl-native/pull/2034#issuecomment-130493229

For some reason a compile error in the OSX job does not fail the job on Travis.

The error isn't from run.sh, it's [this bit](https://travis-ci.org/mapbox/mapbox-gl-native/jobs/75360573#L360):

```
▸ Compiling settings_nsuserdefaults.mm
⌦  /Users/travis/build/mapbox/mapbox-gl-native/platform/darwin/settings_nsuserdefaults.mm:35:28: use of old-style cast [-Werror,-Wold-style-cast]
        userTrackingMode = (MGLUserTrackingMode)uncheckedTrackingMode;
                           ^                    ~~~~~~~~~~~~~~~~~~~~~
** BUILD FAILED **
The following build commands failed:
	CompileC build/osx.build/Debug/osxapp.build/Objects-normal/x86_64/settings_nsuserdefaults.o /Users/travis/build/mapbox/mapbox-gl-native/platform/darwin/settings_nsuserdefaults.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

That's part of `make Xcode/osxapp`, which is implemented [here](https://github.com/mapbox/mapbox-gl-native/blob/5165757c2e05a3bf3e84e67948ff6d86ef25f55e/scripts/main.mk#L101-109).

I'm seeing indications on StackOverflow that `xcodebuild` always returns 0 rather than a proper exit code,  so that's likely the issue. We should switch to [xctool](https://github.com/facebook/xctool) everywhere. It's actually the Travis default, and seems to be better.

--- a/scripts/main.mk
+++ b/scripts/main.mk
@@ -100,7 +100,7 @@ Makefile/%: Makefile/__project__
 
 Xcode/%: Xcode/__project__
 	@printf "$(TEXT_BOLD)$(COLOR_GREEN)* Building target $*...$(FORMAT_END)\n"
-	$(QUIET)$(ENV) xcodebuild \
+	$(QUIET)$(ENV) set -o pipefail && xcodebuild \
 		$(XCODEBUILD_ARGS) \
 		-project ./build/$(HOST_SLUG)/gyp/$(HOST).xcodeproj \
 		-configuration $(BUILDTYPE) \

